weldx/tests/asdf_tests/test_weldx_file.py

```diff
@@ -338,6 +338,79 @@ def test_custom_schema_real_file(tmpdir):
         with pytest.raises(ValueError):
             WeldxFile(custom_schema="no")
 
+    @staticmethod
+    def _get_schema_file(arg, mismatch: bool) -> (str, np.ndarray):
+        data = np.arange(6)
+        if not arg or not isinstance(arg, (list, tuple)):
+            return None, data
+        result = [None, None]
+        shape = (1, -1)
+        schema = """
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "asdf://weldx.bam.de/weldx/schemas/debug/test_shape_validator-0.1.0"
+
+title: test
+type: object
+properties:
+  prop1:
+    tag: "tag:stsci.edu:asdf/core/ndarray-1.*"
+    wx_shape: [{shape}]
+        """
+        d = pathlib.Path(tempfile.gettempdir())
+
+        if isinstance(arg, (list, tuple)):
+            a, b = arg
+            if a is not None:
+                fname = str(d / "a")
+                shape = [2, 3]
+                shape_s = ",".join(str(s) for s in shape)
+                with open(fname, "w+") as fh:
+                    fh.write(schema.format(shape=shape_s))
+                result[0] = fname
+            if b is not None:
+                fname = str(d / "b")
+                shape = [3, 2]
+                shape_s = ",".join(str(s) for s in shape)
+
+                with open(fname, "w+") as fh:
+                    fh.write(schema.format(shape=shape_s))
+                result[1] = fname
```

Comment:
Thank you for adding a testcase for this !

I think there is a way with `asdf` to provide a schema without saving an actual file but I am not sure if it works with our `WeldxFile` wrapper

Here is an example:
https://github.com/asdf-format/asdf/blob/66abc5e4599db381850dddcb293a90c4a9afd999/asdf/tests/test_schema.py#L196-L204

I will look into if we can apply this somehow in our test cases, it should make a lot of things much easier

but this solution is good

Reply:
Thanks for providing the file-free version of schema validation! This looks very useful. Since WeldxFile respects outer `Config` contexts, I think it would be possible to use this pattern. I will try it out quickly.

Reply:
I think what might be an issue is that we only look for schemas as files in our current implementation
https://github.com/BAMWelDX/weldx/blob/ae2dff5c2b4c9e90e95f2b4357b60b291496c157/weldx/asdf/file.py#L221-L230

what we might need is another 'fallback' into `asdf.schema.load_schema()` in the place where we 'resolve' the schema

Reply:
Although it is a nice pattern, the custom_schema argument routing (e.g. tuple unpacking) is only possible within the wrapper. Also AsdfFile expects user schemas to be a file or URI: https://github.com/asdf-format/asdf/blob/66abc5e4599db381850dddcb293a90c4a9afd999/asdf/tests/test_schema.py#L899

Reply:
So we cant use it here?